pytest_operator/plugin.py

```diff
@@ -822,14 +824,22 @@ async def forget_model(
     @staticmethod
     async def _reset(model: Model, allow_failure, timeout: Optional[int] = None):
         # Forcibly destroy applications/machines in case any units are in error.
-        log.info(f"Resetting model {model.info.name}...")
-        for app in model.applications.values():
-            log.info(f"   Destroying application {app.name}")
-            await app.destroy()
+        async def _destroy(entity: str, **kwargs):
+            for key, entity in getattr(model, entity).items():
+                try:
+                    log.info(f"   Destroying {entity} {key}")
+                    await entity.destroy(**kwargs)
+                except DeadEntityException as e:
+                    log.warning(e)
+                    log.warning(f"{entity.title()} already dead, skipping")
```

Comment:
`entity` here is an Application or Machine object right? I don't see a `title` method on those classes so I'm expecting an error here. Is this supposed to be `key.title()`?